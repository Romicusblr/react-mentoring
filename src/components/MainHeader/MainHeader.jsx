import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import ResultCounter from '../ResultCounter';
import RadioSwitch from '../RadioSwitch';
import RadioButton from '../RadioButton';
import style from './MainHeader.module.css';

const MainHeader = ({ quantity, className, ...other }) => (
  <header
    className={classNames(style.mainHeader, className)}
    {...other}
  >
    <ResultCounter quantity={quantity} />
    <RadioSwitch name="sortFilter" title="sort by">
      <RadioButton value="release date" defaultChecked />
      <RadioButton value="rating" />
    </RadioSwitch>
  </header>
);

MainHeader.propTypes = {
  quantity: PropTypes.number,
  className: PropTypes.string,
};

MainHeader.defaultProps = {
  className: '',
  quantity: 0,
};

export default MainHeader;
