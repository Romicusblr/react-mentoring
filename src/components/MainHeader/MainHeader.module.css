.mainHeader {
  display: flex;
  height: 5rem;
  align-items: center;
  justify-content: space-between;
  padding: 0 var(--padding-header);
}
