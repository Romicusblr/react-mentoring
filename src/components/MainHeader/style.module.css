.mainHeader {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: .75rem var(--padding-header);
}
