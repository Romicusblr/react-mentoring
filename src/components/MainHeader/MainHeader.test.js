import MainHeader from '.';

describe('MainHeader', () => {
  it('should match to snapshot', () => {
    const snap = shallow(<MainHeader />);
    expect(snap).toMatchSnapshot();
  });
});
