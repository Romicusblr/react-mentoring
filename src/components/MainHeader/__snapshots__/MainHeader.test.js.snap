// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHeader should match to snapshot 1`] = `
<header
  className="mainHeader"
>
  <ResultCounter
    className=""
    quantity={0}
  />
  <RadioSwitch
    className=""
    name="sortFilter"
    title="sort by"
  >
    <RadioButton
      className=""
      defaultChecked={true}
      name="radio"
      value="release date"
    />
    <RadioButton
      className=""
      defaultChecked={false}
      name="radio"
      value="rating"
    />
  </RadioSwitch>
</header>
`;
